src/hardware/serialport/softmodem.cpp

```diff
@@ -150,6 +150,15 @@ CSerialModem::CSerialModem(const uint8_t port_idx, CommandLine *cmd)
 		        GetPortNumber(), telnet_mode ? "enabled" : "disabled");
 	}
 
+	// Get the connect speed to report
+	if (getUintFromString("baudrate:", val, cmd)) {
+		LOG_MSG("SERIAL: Port %" PRIu8 " will report baud rate %d",
+			GetPortNumber(), val);
+	} else {
+		val = 57600;
+	}
+	snprintf(connect_string, 255, "CONNECT %d", val);
```

Comment:
I know this is a nitpick, but baud rates for serial back in the DOS days never typically exceeded 115200.  I know there were some specialized hardware that could exceed that, but from my research, it looks like 256000 was probably the limit.  Though I think with modern devices it can go extremely high.  I don't see the need though to exceed 6 characters.  Probably anything exceeding that in the config option is a mistake and should either be ignored or truncated.  I could nitpick further and check for numeric only values, but in the end it's not really important for what it does.

Reply:
Good catch, @bmunger. Looks like any positive value is currently permitted.
There were a handful of common CONNECT values ranging from 9600, 19200, to 38400 and so on. 

Although, I do recall when line conditions were bad, getting marginal rates too - like instead of the full 19.2 kbaud, connecting at 16.8 kbaud.

Does it make sense to clamp to the nearest "spec" value or should we permit any value with a 57.6 kbaud cap?

Reply:
I remember those in between rates too.  I have been thinking about this, and I'm not sure what value it would actually bring for the added complexity, given that it really doesn't have any effect on the actual connection speed.  I'm all in favor of accuracy when it makes sense.  Maybe that would be best left for a future change if the baud rate actually affects the connection speed.  I know the 16650 uart had a limit of 115200 bps, but of course this is part of the soft modem code.  The maximum speed a modem could reach I think was 53333 bps.

Reply:
Sounds reasonable to not over-restrict this if it's mostly superficial anyway. Maybe just the a cap then, and allow any value in between 0 and the cap?

Reply:
Ok, pushed a commit that allows user-specified non-zero positive integer baudrate <= 57600, otherwise will default to 57600.